gips/0080-substreams-on-the-network.md

```diff
@@ -0,0 +1,112 @@
+
+### Use of contracts
+
+References:
+
+ - [DataEdge addresses](https://github.com/graphprotocol/contracts/blob/main/packages/data-edge/addresses.js)
+- The relevant contract sources are `DataEdge` ([DataEdge.sol](https://github.com/graphprotocol/contracts/blob/main/packages/data-edge/contracts/DataEdge.sol)) and `EventfulDataEdge` ([EventfulDataEdge.sol](https://github.com/graphprotocol/contracts/blob/main/packages/data-edge/contracts/EventfulDataEdge.sol)). `EventfulDataEdge` emits an event upon registration, while `DataEdge` does not.
+
+The exact format of the published data will be documented in the Substreams documentation and implemented in the command-line tools, as well as in the Substreams module to keep the front-end up-to-date.
+
+
+### CLI for Registration
+
+To enable on-chain registration of Substreams providers for The Graph Market, network payments tools will be augmented (or the `substreams` command-line) with a new `register` command.  This command will publish provider information to DataEdge, signed by the operator's key.
+
+```
+substreams network register --operator-wallet 0x123123123 --operator-priv-key-file my.key --service substreams --endpoint mainnet.eth.superbob.com --logo ./mylogo.png --name Pinax Networks --network ethereum
+```
+
+This command allows providers to register their service, specifying their operator wallet, private key file, the service type (`substreams`), the endpoint URL, logo, name, and the supported network. The registration data is signed by the operator's key.  An update to an existing registration can be achieved by re-running the command with updated parameters.  A corresponding `unregister` or `revoke` command would be added to allow providers to take down their service from the front-end.
+
+A background process will continuously monitor DataEdge for new registrations and updates.  Upon detecting a new registration, the system will perform health checks against the provided endpoint (e.g., calling `/health`, `/head`, or `/info`) to verify the service's availability and type. Only after successful health checks will the provider be added to the `providers.json` file, which is used to populate the front-end UI.  This ensures that only active and validated providers are displayed to users.  The system will incorporate block time checks to prevent displaying stale registrations until the block height is sufficiently close to the chain tip.
+
+
+## Feature Matrix
+
+This GIP does _not_ propose enabling indexing rewards for Substreams just yet. It is the first step towards rolling out all what is necessary to eventually get there.  This is contingent on data determinism assurances and Indexer readiness. Future GIPs will propose expanding indexing rewards.
```

Comment:
As mentioned above - enabling indexing rewards would also require figuring out what activity would be rewarding (is it a firehose POI?) and might require smart contracts changes too

Reply:
yes, so perhaps we detail that in a future GIP